Refactor beneficios implemenatando el store procedure para el post de beneficios

Fabian-mo29 opened the pull request:
[SCRUM-12](https://infinipay.atlassian.net/browse/SCRUM-12)
Se crea el store procedure para crear beneficios usando la nueva arquitectura.

24danielcr committed:
Refactor de beneficios asignados por empleado.

Parte del refactor de beneficios en la compañia también hecho.

24danielcr committed:
Refactor de beneficios asignados por empleado.

Parte del refactor de beneficios en la compañia también hecho.

24danielcr committed:
Refactor de beneficios asignados por empleado.

Parte del refactor de beneficios en la compañia también hecho.

24danielcr committed:
Refactor de beneficios asignados por empleado.

Parte del refactor de beneficios en la compañia también hecho.

Fabian-mo29 committed:
Implementacion de el store procedure y la refactorizacion de crear beneficios con la nueva arquitectura

Fabian-mo29 committed:
Implementacion de el store procedure y la refactorizacion de crear beneficios con la nueva arquitectura

Fabian-mo29 committed:
Implementacion de el store procedure y la refactorizacion de crear beneficios con la nueva arquitectura

Fabian-mo29 committed:
Implementacion de el store procedure y la refactorizacion de crear beneficios con la nueva arquitectura

Fabian-mo29 committed:
Arreglo para que cree los beneficios desde el frontend

Fabian-mo29 committed:
Arreglo para que cree los beneficios desde el frontend

Fabian-mo29 committed:
Arreglo para que cree los beneficios desde el frontend

Fabian-mo29 committed:
Arreglo para que cree los beneficios desde el frontend

Fabian-mo29 committed:
Se agrega el store procedure que crea los beneficios ademas del fix a los valores vmodel de registrarbeneficios

Fabian-mo29 committed:
Se agrega el store procedure que crea los beneficios ademas del fix a los valores vmodel de registrarbeneficios

Fabian-mo29 committed:
Se agrega el store procedure que crea los beneficios ademas del fix a los valores vmodel de registrarbeneficios

Fabian-mo29 committed:
Se agrega el store procedure que crea los beneficios ademas del fix a los valores vmodel de registrarbeneficios

braunnymadrigal committed:
agregadas columnas necesarias para utilizar las apis externas

24danielcr committed:
Front-end registar horas

https://infinipay.atlassian.net/browse/SCRUM-110?atlOrigin=eyJpIjoiOWFhNzQyZGViOTRhNDIzYzk5YWUxZTY3MmUzMDY1NDciLCJwIjoiaiJ9

24danielcr committed:
Queries para obtener horas de del empleado y tipo de contrato.

https://infinipay.atlassian.net/browse/SCRUM-111?atlOrigin=eyJpIjoiNThhNWM3OWE4YmM5NDZiNTljZTRlMDA5NTg2M2Y0ZWQiLCJwIjoiaiJ9

24danielcr committed:
Registrar horas y repositorio.

https://infinipay.atlassian.net/browse/SCRUM-112?atlOrigin=eyJpIjoiZGMzYjg3N2M2ZTIzNGQ2OTk0MjljMmU3YjUxMzU2MjgiLCJwIjoiaiJ9

24danielcr committed:
Creación de tabla para API externas

24danielcr committed:
Se agrega tabla de API externa

24danielcr committed:
Bugfix: Arreglo fechas registrando mal por tiempos

Fabian-mo29 committed:
Modificaciones en employee creando el command y la funcionalidad de editar datos de empleado

Fabian-mo29 committed:
Modificaciones en employee creando el command y la funcionalidad de editar datos de empleado

Fabian-mo29 committed:
Implementacion del endpoint para el get user by id

Fabian-mo29 committed:
Implementacion del endpoint para el get user by id

Commit:
Store procedure para la creacion de un empleado en empresa

Commit:
Cambiar el tipo de contrato de empleado

Commit:
Casos de pruebas para el calculo de impuesto de renta

Commit:
Inyectar dependencias

Fabian-mo29 committed:
Correccion del manejo de password, implementacion del trigger y actualizacion de datos de Auditoria

Fabian-mo29 committed:
Correccion del manejo de password, implementacion del trigger y actualizacion de datos de Auditoria

Fabian-mo29 committed:
Frontend del formulario para el update de los empleados

Fabian-mo29 committed:
Frontend del formulario para el update de los empleados

mafer1404 committed:
Cambiar el tipo de empleado elegible

mafer1404 committed:
Rename SQLQuery026 to SQLQuery026.sql

24danielcr committed:
Cambio en manejo de error en obtención de horas registradas si no existe.

24danielcr committed:
Arreglo reincio de horas obtenidas por semana

Commit:
Agregar el nombre del empleado en la planilla resultante

24danielcr committed:
Arreglo columna empleadoElegible para recibir más char

mafer1404 committed:
Merge pull request #54 from braunnymadrigal/sqlconsultas

Sqlconsultas

24danielcr committed:
Refactor de beneficios de empleado.

Mejor manejo de errores, separación de responsabilidades, dependency inyection

24danielcr committed:
Refactor de beneficios de empleado.

Mejor manejo de errores, separación de responsabilidades, dependency inyection

24danielcr committed:
Refactor de beneficios de empleado.

Mejor manejo de errores, separación de responsabilidades, dependency inyection

24danielcr committed:
Refactor de beneficios de empleado.

Mejor manejo de errores, separación de responsabilidades, dependency inyection

Commit:
Agregar rentTax al modelo

braunnymadrigal committed:
Merge remote-tracking branch 'origin/unitTestingRentTax' into testingGrossSalary

merge con unitTestRentTax

24danielcr committed:
Arreglo pantalla atorada en cargando.

braunnymadrigal committed:
creado el primer test para bruto

braunnymadrigal committed:
agregado suficiente unit testing para el compute week salary

braunnymadrigal committed:
agregados los tests para el quincenal

braunnymadrigal committed:
agregados los tests para el mensual

braunnymadrigal committed:
eliminados los comentarios redudantes

braunnymadrigal reviewed (changes requested):
Denegado, primero resuelva los conflictos.

braunnymadrigal committed:
creada la clase para testear el impuesto de la CCSS

24danielcr committed:
Cambio en orden de UseCors para obtener mensajes de error más específicos tras falta de permisos. Manejor

24danielcr committed:
Mejor manejo de errores en front end

24danielcr committed:
Modularización del código

braunnymadrigal committed:
agregada una cantidad suficiente de unit tests para CCSS

braunnymadrigal committed:
creada la clase para hacer unit tests de Deduction

braunnymadrigal committed:
Merge pull request #56 from braunnymadrigal/testingGrossSalary

Unit tests para el cálculo de salario bruto

braunnymadrigal committed:
Merge pull request #57 from braunnymadrigal/testingCCSS

 Unit tests para el cálculo del impuesto CCSS

Fabian-mo29 committed:
Fix a los campos del form de tipo de contrato

Fabian-mo29 committed:
Fix a los campos del form de tipo de contrato

braunnymadrigal committed:
update para que se agregue al menos una deduccion en el constructor

24danielcr committed:
Arreglo Conflictos

24danielcr committed:
Cambio de connection string a default.

mafer1404 committed:
Modificar que idDeduccion pueda ser nulo

Esto para que no haya que crear una deduccion de rentTax y ccssTax, solamente se insertan en DetallePago

braunnymadrigal committed:
agregados los metodos para testear los distintos tipos de deduccion por motivo de beneficio

braunnymadrigal committed:
Merge branch 'develop' into testingBenefitsDeductions

merge con develop completado con exito

Fabian-mo29 committed:
Merge branch 'editar_empleado' of https://github.com/braunnymadrigal/Infinipay-CO into editar_empleado

24danielcr committed:
Store procedure crear empleado.

Fabian-mo29 committed:
Arreglos despues del rebase

Fabian-mo29 committed:
Merge pull request #53 from braunnymadrigal/editar_empleado

Update de empleado backend y frontend

24danielcr committed:
Se agrega asociación a registro de empresa.

24danielcr committed:
Refactor the repositorio de empleador para nuevo store procedure.

braunnymadrigal committed:
Merge pull request #59 from braunnymadrigal/testingBenefitsDeductions

 Unit tests para el cálculo de deducciones por motivo de beneficio asignado

Fabian-mo29 committed:
Refactorizacion y cambios

Fabian-mo29 committed:
Refactorizacion y cambios

Fabian-mo29 committed:
Refactorizacion y cambios

Fabian-mo29 committed:
Refactorizacion y cambios

Fabian-mo29 committed:
Arreglos y refactor despues de el rebase

Fabian-mo29 committed:
Arreglos y refactor despues de el rebase

Fabian-mo29 committed:
Merge branch 'refactor_beneficios' of https://github.com/braunnymadrigal/Infinipay-CO into refactor_beneficios

Commit:
Endpoints para generar y mostrar planilla

Fabian-mo29 committed:
Correccion en la validacion de nombre de beneficio duplicado

Fabian-mo29 committed:
Correccion en la validacion de nombre de beneficio duplicado

Commit:
Clase orquestradora para calcular planillas de empleados con salario bruto, impuesto sobre la renta, CCSS y deducciones

Commit:
Modelo de request de planilla

Commit:
Modelo de resultados de planilla

Commit:
Repositorio para guardar la planilla y obtener sus datos para mostrar resultados

Commit:
Agregar rutas para generar planilla y mostrar resultados

Commit:
Front end para mostrar generar planilla

Commit:
Front end para mostrar resultados de planilla

Commit:
Ajustar logica de calculo de impuesto de renta

Commit:
Eliminar archivo no usado

Commit:
Arreglar pruebas de RentTax y agregar mas casos

Commit:
Casos de prueba para PayrollEmployees

Commit:
Agregar Moq al proyecto

Commit:
Eliminar comentarios redundantes y clean code

mafer1404 committed:
Merge branch 'develop' into generarPlanilla

mafer1404 committed:
Actualizar modelo logico 

Nuevas columnas en algunas tablas

Fabian-mo29 committed:
Correcciones para los endpoint y los storeprocedures

Fabian-mo29 committed:
Correcciones para los endpoint y los storeprocedures

24danielcr committed:
Merge pull request #60 from braunnymadrigal/BugFixRegistrarEmpresa

Bug fix registrar empresa

Fabian-mo29 committed:
Arreglo a el sp de storeBenefits y al frontend para el updateBenefits

Fabian-mo29 committed:
Arreglo a el sp de storeBenefits y al frontend para el updateBenefits

24danielcr committed:
Arreglo a link y parametros para obtener y asignar beneficios de empleado.

24danielcr committed:
Arreglo a link y parametros para obtener y asignar beneficios de empleado.

24danielcr committed:
Merge branch 'refactor_beneficios' of https://github.com/braunnymadrigal/Infinipay-CO into refactor_beneficios

24danielcr committed:
Merge branch 'develop' into registrar_horas

24danielcr committed:
Merge pull request #58 from braunnymadrigal/registrar_horas

Registrar horas

mafer1404 committed:
Merge branch 'develop' into generarPlanilla

mafer1404 committed:
Merge pull request #61 from braunnymadrigal/generarPlanilla

Generar planilla back end y front end

mafer1404 committed:
Merge pull request #62 from braunnymadrigal/unitTestingPayrollEmployee

Unit testing para obtener los empleados de la planilla

Commit:
Merge branch 'develop' of https://github.com/braunnymadrigal/Infinipay-CO into develop

Commit:
Eliminar duplicado en modelo

Commit:
Agregar fulName ya que es parte del modelo

Commit:
Arregar fullName ya que es parte del modelo

Commit:
Eliminar comentarios

Commit:
Evitar nombres repetidos en fullName

Fabian-mo29 committed:
Fix de el repository y de los assigned benefits tras el rebase

Fabian-mo29 committed:
Merge branch 'refactor_beneficios' of https://github.com/braunnymadrigal/Infinipay-CO into refactor_beneficios

Fabian-mo29 committed:
Fix despues de rebase

Fabian-mo29 commented:
[SCRUM-62](https://infinipay.atlassian.net/browse/SCRUM-62)
Se le suma el update de beneficios mediante otro store procedure
